### PR TITLE
fix: fix cjs live binding import name sourcemap

### LINKED ISSUE #2794: [Bug]: wrong source map for cjs live binding import name
### Reproduction link or steps

reproduction: https://github.com/hi-ogawa/reproductions/tree/main/rolldown-source-map-cjs-binding (same one on stackblitz  https://stackblitz.com/edit/github-s9e2ts?file=dist%2Fmain1.js)

When named import (here `throwError`) is transformed into live binding property access (here `require_lib.throwError`), it loses source mapping: [sourcemap viz](https://evanw.github.io/source-map-visualization/#NTU3ACJ1c2Ugc3RyaWN0IjsKY29uc3QgcmVxdWlyZV9saWIgPSByZXF1aXJlKCcuL2xpYi1DVTNQT0tEUi5qcycpOwoKLy8jcmVnaW9uIHNyYy9tYWluMS5qcwpmdW5jdGlvbiBtYWluKCkgewoJcmVxdWlyZV9saWIudGhyb3dFcnJvcigpOwp9Cm1haW4oKTsKCi8vI2VuZHJlZ2lvbgovLyMgc291cmNlTWFwcGluZ1VSTD1kYXRhOmFwcGxpY2F0aW9uL2pzb247Y2hhcnNldD11dGYtODtiYXNlNjQsZXlKMlpYSnphVzl1SWpvekxDSm1hV3hsSWpvaWJXRnBiakV1YW5NaUxDSnVZVzFsY3lJNld5SWlYU3dpYzI5MWNtTmxjeUk2V3lJdUxpOXpjbU12YldGcGJqRXVhbk1pWFN3aWMyOTFjbU5sYzBOdmJuUmxiblFpT2xzaWFXMXdiM0owSUhzZ2RHaHliM2RGY25KdmNpQjlJR1p5YjIwZ1hDSXVMMnhwWWk1cWMxd2lYRzVjYm1aMWJtTjBhVzl1SUcxaGFXNG9LU0I3WEc0Z0lIUm9jbTkzUlhKeWIzSW9LVnh1ZlZ4dVhHNXRZV2x1S0NsY2JpSmRMQ0p0WVhCd2FXNW5jeUk2SWpzN096dEJRVVZCTEZOQlFWTXNUMEZCVHp0QlFVTmtMRU5CU0VaQkxIZENRVWRqTzBGQlEySTdRVUZGUkN4TlFVRk5JbjA9MjUxAHsidmVyc2lvbiI6MywiZmlsZSI6Im1haW4xLmpzIiwibmFtZXMiOlsiIl0sInNvdXJjZXMiOlsiLi4vc3JjL21haW4xLmpzIl0sInNvdXJjZXNDb250ZW50IjpbImltcG9ydCB7IHRocm93RXJyb3IgfSBmcm9tIFwiLi9saWIuanNcIlxuXG5mdW5jdGlvbiBtYWluKCkge1xuICB0aHJvd0Vycm9yKClcbn1cblxubWFpbigpXG4iXSwibWFwcGluZ3MiOiI7Ozs7QUFFQSxTQUFTLE9BQU87QUFDZCxDQUhGQSx3QkFHYztBQUNiO0FBRUQsTUFBTSJ9)

image

This can cause wrong stacktrace like in this example:

```
$ node --enable-source-maps ./dist/main1.js
Trace: yay
    at Object.throwError (<root>/src/lib.js:2:11)
    at main (<root>/src/main1.js:1:1)  // 👈👈 expected: main1.js:4:3
    at Object.<anonymous> (<root>/src/main1.js:7:1)
    ...
```

### What is expected?

mapping of `throwError` at `main1.js:4:3`

### What is actually happening?

mapping of `throwError` at `main1.js:1:1`

### System Info

```Shell
System:
    OS: Linux 6.11 Arch Linux
    CPU: (16) x64 12th Gen Intel(R) Core(TM) i7-12650H
    Memory: 20.03 GB / 31.06 GB
    Container: Yes
    Shell: 5.2.37 - /usr/bin/bash
  Binaries:
    Node: 20.18.0 - ~/.volta/tools/image/node/20.18.0/bin/node
    npm: 10.8.2 - ~/.volta/tools/image/node/20.18.0/bin/npm
    pnpm: 9.13.2 - ~/.volta/bin/pnpm
    bun: 1.1.29 - ~/.volta/bin/bun
  Browsers:
    Chromium: 130.0.6723.91
  npmPackages:
    rolldown: nightly => 0.14.0-snapshot-0e16a9a-20241119003440
```

### Any additional comments?

App format had a similar issue and while debugging that, I found a reproduction in cjs format.

### PR DESCRIPTION
**Pull request by hi-ogawa**
### Description

- Closes https://github.com/rolldown/rolldown/issues/2794

To verify

```
$ echo '{"type":"commonjs"}' > crates/rolldown/tests/rolldown/topics/live_bindings/sourcemap_cjs/dist/package.json
$ node --enable-source-maps crates/rolldown/tests/rolldown/topics/live_bindings/sourcemap_cjs/dist/main1.js
Trace: shared
    at Object.shared (.../sourcemap_cjs/shared.js:2:11)
    at main (.../sourcemap_cjs/main1.js:4:3)
    at Object.<anonymous> (.../sourcemap_cjs/main1.js:7:1)
    ...
```

### CODE REVIEW

File: crates/rolldown/tests/rolldown/topics/live_bindings/sourcemap_cjs/artifacts.snap [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+# Sourcemap Visualizer
+
+```
+- ../main1.js
+(2:0-2:9) "\nfunction" --> (4:0-4:9) "\nfunction"
+(2:9-2:16) " main()" --> (4:9-4:16) " main()"
+(2:16-3:2) " {\n " --> (4:16-5:0) " {"
+(3:2-0:0) "" --> (5:0-5:1) "\n"
+(0:0-3:2) "import {shared} from './shared.js'\n\nfunction main() {\n " --> (5:1-5:16) "\trequire_shared"
```
Comment (hi-ogawa) [inline crates/rolldown/tests/rolldown/topics/live_bindings/sourcemap_cjs/artifacts.snap]: `(0:0-3:2) -> (5:1-5:16)` looks wrong, but not sure what's causing this.

File: crates/rolldown/tests/rolldown/topics/live_bindings/sourcemap_cjs/artifacts.snap [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+(3:2-3:10) " shared(" --> (5:16-5:24) ".shared("
```
Comment (hi-ogawa) [inline crates/rolldown/tests/rolldown/topics/live_bindings/sourcemap_cjs/artifacts.snap]: Previously this mapping `(3:2-3:10) -> (5:16-5:24)` didn't exist.

File: crates/rolldown/tests/rolldown/topics/live_bindings/sourcemap_cjs/_config.json
Comment (hyf0) [inline crates/rolldown/tests/rolldown/topics/live_bindings/sourcemap_cjs/_config.json:1]: I'll suggest to create `topics/sourcemap` for this test. `topics/live_bindings` is for ensure the behavior of live binding.

File: crates/rolldown/tests/rolldown/topics/live_bindings/sourcemap_cjs/artifacts.snap (hunk 1 above)
Comment (hyf0) [reply crates/rolldown/tests/rolldown/topics/live_bindings/sourcemap_cjs/artifacts.snap]: Fee free to add cases in https://github.com/rolldown/rolldown/issues/2737.

I'll notice this too. But it's too cumbersome to solve them now. The issue chain is too long.

Comment (underfin) [reply crates/rolldown/tests/rolldown/topics/live_bindings/sourcemap_cjs/artifacts.snap]: Here caused by `SourcemapVisualizer`, the `\trequire_shared` corresponding token should remapping to `import {shared} from './shared.js'\n`. Here i meet some issues with read content cross lines at two tokens, so here print has some misleading.

Comment (hyf0) [reply crates/rolldown/tests/rolldown/topics/live_bindings/sourcemap_cjs/artifacts.snap]: @underfin  So there's also bug of `SourcemapVisualizer`? I don't think we need to fix it now, but would be good to fix it. Anyway, I'll record this in https://github.com/rolldown/rolldown/issues/2737.

Comment (underfin) [reply crates/rolldown/tests/rolldown/topics/live_bindings/sourcemap_cjs/artifacts.snap]: Yeah. Let us track to it.

File: crates/rolldown/tests/rolldown/topics/live_bindings/sourcemap_cjs/_config.json
Comment (hi-ogawa) [reply crates/rolldown/tests/rolldown/topics/live_bindings/sourcemap_cjs/_config.json:1]: There is `crates/rolldown/tests/rolldown/sourcemap` which is outside of `topics` directory. Should I create `tests/rolldown/topics/sourcemap` or put it in `tests/rolldown/sourcemap` for now?

**Review by hyf0 — APPROVED** [review]
Thanks!

**Pull request merged**